src/main/java/org/embulk/input/zendesk/clients/ZendeskRestClient.java

```diff
@@ -0,0 +1,282 @@
+package org.embulk.input.zendesk.clients;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.google.common.util.concurrent.RateLimiter;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.embulk.config.ConfigException;
+import org.embulk.input.zendesk.ZendeskInputPlugin.PluginTask;
+import org.embulk.input.zendesk.models.ZendeskException;
+import org.embulk.input.zendesk.utils.ZendeskConstants;
+import org.embulk.input.zendesk.utils.ZendeskUtils;
+import org.embulk.spi.Exec;
+import org.embulk.spi.util.RetryExecutor;
+import org.slf4j.Logger;
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+import static org.apache.http.protocol.HTTP.CONTENT_TYPE;
+import static org.embulk.spi.util.RetryExecutor.retryExecutor;
+
+import java.io.IOException;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class ZendeskRestClient
+{
+    private static final int CONNECTION_TIME_OUT = 240000;
+
+    private static final Logger logger = Exec.getLogger(ZendeskRestClient.class);
+
+    private static RateLimiter rateLimiter;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    static {
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    public ZendeskRestClient()
+    {
+    }
+
+    private String sendRequest(final String url, final PluginTask task) throws ZendeskException
+    {
+        try {
+            HttpClient client = createHttpClient();
+            HttpRequestBase request = createGetRequest(url, task);
```

Comment:
I'm wondering should we refactor here, the name of method is quite general `sendRequest` but inside of the implementation it always creates the `GET` method, that leads the `request.getMethod()` printed message below sounds redundant because it is always `GET`.